jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/SelectableChannelEndPoint.java

```diff
@@ -148,6 +160,18 @@ public boolean isOpen()
         return _channel.isOpen();
     }
 
+    @Override
+    public void onOpen()
+    {
+        super.onOpen();
+        /* Cache the local and remote Socket Address in this EndPoint class.
+         * as they are inaccessible once the EndPoint is closed.
+         * This helps RequestLog to access the remote and local addresses, even on a closed connection.
+         */
+        _localSocketAddress = getLocalSocketAddress();
+        _remoteSocketAddress = getRemoteSocketAddress();
```

Comment:
These are weirdly circular as `getLocalSocketAddress()` is used to set `_localSocketAddress`, but it may return `_localSocketAddress`.

Just put the logic to get them in one spot (preferably constructor with final fields, but if not then here with normal fields) and just return those fields from the getter methods.